diff --git a/node_modules/expo-eas-observe/ios/AppMetricsModule.swift b/node_modules/expo-eas-observe/ios/AppMetricsModule.swift
index b565664..522c355 100644
--- a/node_modules/expo-eas-observe/ios/AppMetricsModule.swift
+++ b/node_modules/expo-eas-observe/ios/AppMetricsModule.swift
@@ -3,25 +3,32 @@ import ExpoModulesCore
 internal let logger = Logger(logHandlers: [createOSLogHandler(category: Logger.EXPO_LOG_CATEGORY)])
 
 public final class ExpoAppMetricsModule: Module {
+  private func refreshUpdateIdIfNeeded(_ updateId: String?) {
+    guard let updateId = updateId else { return }
+    AppMetricsActor.isolated {
+      let current = AppInfo.current
+      if updateId != current.updateId {
+        let patched = AppInfo(
+          appId: current.appId,
+          appName: current.appName,
+          appVersion: current.appVersion,
+          buildNumber: current.buildNumber,
+          updateId: updateId
+        )
+        AppInfo.current = patched
+        logger.debug("Refreshed AppInfo with new updateId = \(updateId)")
+        AppMetrics.storage.currentEntry.app = patched
+      }
+    }
+  }
+
   public func definition() -> ModuleDefinition {
     Name("ExpoAppMetrics")
 
     OnCreate {
+      refreshUpdateIdIfNeeded(AppInfo.getLaunchedUpdateId())
       AppMetricsActor.isolated {
         AppMetrics.mainSession.startMonitoringFrames()
-
-        let current = AppInfo.current
-        if current.updateId == nil, let updateId = AppInfo.getLaunchedUpdateId() {
-          let patched = AppInfo(
-            appId: current.appId,
-            appName: current.appName,
-            appVersion: current.appVersion,
-            buildNumber: current.buildNumber,
-            updateId: updateId
-          )
-          AppInfo.current = patched
-          AppMetrics.storage.currentEntry.app = patched
-        }
       }
     }
 
